lib/src/http/response.dart

```diff
@@ -96,6 +101,34 @@ class Response extends BaseResponse {
     _body =
         new HttpBody.fromString(contentType, body, fallbackEncoding: encoding);
   }
+
+  /// Create a new [Response] using all the values from this instance except
+  /// for the parameters specified.
+  Response replace(
+      {List<int> bodyBytes,
+      String bodyString,
+      int status,
+      String statusText,
+      Map<String, String> headers}) {
+    if (status == null) {
```

Comment:
it could probably be cleaned up a bit with null aware operators, but w_transport is still compatible (I think) with Dart SDK >=1.9.0 (https://github.com/Workiva/w_transport/blob/master/pubspec.yaml#L40-L41) so unless we really need to use something from a later SDK release or we find that we're not actually compatible that far back, I'd rather not change that.